### DISCUSSION TITLE
Pratt parsing without expecting an operator

### DISCUSSION THREAD

**Discussion by FilipToth** [pest-parser/pest#808] [Q&A]
I need my pratt parser to simply consume a list of tokens without there specifically being an operation involved. Then run separate parse/eval methods on the subtokens. I've made a simple example that just takes a list of numbers.
```
module = { SOI ~ root ~ EOI }
root = { (num)* }

num = @{ ASCII_DIGIT+ }

WHITESPACE = _{ " " | "\t" }
```
I still need operations on other tokens, I just don't want operators to take place between the subtokens of the `root` token. I am currently getting the following error: `Expected operator, found num(4, 8)` for input: `175 1284`. Is there any way to do this using the pratt parser?

**Comment by tomtau** [comment]
AFAIK the Pratt parser should be passed operators -- CC @segeljakt ?

**Reply by segeljakt** [reply]
Yeah, you can pass the output of pest to the pratt parser to handle precedence/associativity of infix/prefix/postfix operators.

More info here: https://pest.rs/book/precedence.html
